### PR TITLE
fix: Issues with Mirror Node Pinger

### LINKED ISSUE #2167: bug: Mirror Node monitor's pinger is not connecting correctly
I set the pinger tps to 25 to test the Mirror Node memory consumption.  When investigating, I noticed that the monitor is erroring out.  

We need to update our test harness to make sure that the monitor is not failing with the pinger enabled, and we need to fix this issue.

My output:
```
2025-06-13T19:24:14.606Z  INFO main o.h.m.m.MonitorApplication The following 1 profile is active: "kubernetes"
2025-06-13T19:24:20.134Z  INFO main o.h.m.m.s.r.RestApiClient Connecting to mirror node http://mirror-rest:80/api/v1
2025-06-13T19:24:20.313Z  INFO main o.h.m.m.p.NodeSupplier Starting node validation
2025-06-13T19:24:20.612Z  INFO validator-1 o.h.m.m.p.NodeSupplier Refreshing node list
2025-06-13T19:24:20.612Z  INFO validator-2 o.h.m.m.p.NodeSupplier Validating node NodeProperties(accountId=0.0.3, certHash=null, host=10.30.0.13, nodeId=0, port=50211)
2025-06-13T19:24:21.022Z  INFO main o.h.m.m.s.g.GrpcClientSDK Connecting 1 clients to mirror-grpc:5600
2025-06-13T19:24:21.210Z  INFO main o.h.m.m.p.g.CompositeTransactionGenerator Activated scenario: PublishScenarioProperties(super=ScenarioProperties(duration=PT2562047H47M16.854775807S, enabled=true, limit=9223372036854775807, name=pin
ger, retry=ScenarioProperties.RetryProperties(maxAttempts=1, maxBackoff=PT1S, minBackoff=PT0.5S)), logResponse=false, maxMemoLength=100, properties={amount=1, maxTransactionFee=10000, recipientAccountId=0.0.55, senderAccountId=0.0.2, t
ransferTypes.0=CRYPTO, topicId=${topic.ping}}, receiptPercent=1.0, recordPercent=0.0, timeout=PT13S, tps=25.0, type=CRYPTO_TRANSFER)
2025-06-13T19:24:21.499Z  INFO single-1 o.h.m.m.c.MonitorConfiguration Starting publisher flow
2025-06-13T19:24:21.508Z  INFO parallel-39 o.h.m.m.c.MonitorConfiguration Starting subscribe flow
2025-06-13T19:24:21.716Z  INFO single-1 o.h.m.m.p.NodeSupplier Refreshing node list
2025-06-13T19:24:23.512Z  WARN validator-2 o.h.m.m.p.NodeSupplier Unable to validate node NodeProperties(accountId=0.0.3, certHash=null, host=10.30.0.13, nodeId=0, port=50211):  com.hedera.hashgraph.sdk.PrecheckStatusException: Hedera
transaction `0.0.2@1749842651.499000853` failed pre-check with the status `INVALID_NODE_ACCOUNT`
        at com.hedera.hashgraph.sdk.Executable$GrpcRequest.mapStatusException(Executable.java:968)
        at com.hedera.hashgraph.sdk.Executable.execute(Executable.java:451)
        at com.hedera.hashgraph.sdk.Transaction.execute(Transaction.java:36)
        at com.hedera.hashgraph.sdk.Executable.execute(Executable.java:353)
        at com.hedera.hashgraph.sdk.Transaction.execute(Transaction.java:36)
        at org.hiero.mirror.monitor.publish.NodeSupplier.validateNode(NodeSupplier.java:186)
        at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106)
        at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.runAsync(FluxPublishOn.java:446)
        at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.run(FluxPublishOn.java:533)
        at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84)
        at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)

2025-06-13T19:24:23.608Z  INFO validator-2 o.h.m.m.p.NodeSupplier 0 of 1 nodes are functional
2025-06-13T19:24:23.609Z  WARN hedera-sdk-0 o.h.m.m.e.ExpressionConverterImpl Retry attempt #1 after failure: com.hedera.hashgraph.sdk.PrecheckStatusException: Hedera transaction `0.0.2@1749842651.807000918` failed pre-check with the status `INVALID_NODE_ACCOUNT`
2025-06-13T19:24:23.609Z  INFO validator-2 o.h.m.m.p.NodeSupplier Retrying in 120000 ms
2025-06-13T19:24:24.805Z  WARN expression-2 o.h.m.m.e.ExpressionConverterImpl Retry attempt #2 after failure: java.lang.IllegalArgumentException: No valid nodes available
2025-06-13T19:24:25.305Z  INFO main o.h.m.m.MonitorApplication Started MonitorApplication in 15.584 seconds (process running for 16.85)
2025-06-13T19:24:25.309Z  INFO scheduling-2 o.h.m.m.s.SubscribeMetrics No subscribers
2025-06-13T19:24:25.309Z  INFO scheduling-1 o.h.m.m.p.PublishMetrics No publishers
2025-06-13T19:24:26.429Z  WARN expression-2 o.h.m.m.e.ExpressionConverterImpl Retry attempt #3 after failure: java.lang.IllegalArgumentException: No valid nodes available
2025-06-13T19:24:31.922Z  WARN expression-2 o.h.m.m.e.ExpressionConverterImpl Retry attempt #4 after failure: java.lang.IllegalArgumentException: No valid nodes available
2025-06-13T19:24:35.314Z  INFO scheduling-4 o.h.m.m.p.PublishMetrics No publishers
2025-06-13T19:24:35.314Z  INFO scheduling-1 o.h.m.m.s.SubscribeMetrics No subscribers
```

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
## Description

This PR fixes some issues with the mirror node pinger:
 - Pinger was always enabled, even if the `--pinger` flag was not passed during deployment.
 - Pinger wasn't working with non-zero realm and shard deployments.
 - It was not working properly in a multi-cluster setup.

Extends the `MirrorNodeTest` class with methods to assert the pinger is working correctly if enabled.

### Related Issues

* Closes #2167 
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

**Review by andrewb1269hg — APPROVED** [review]
Review and approve:
`.github/workflows/script/gcs_test.sh`
`.github/workflows/script/launch_network.sh`

File: resources/mirror-node-values-hedera.yaml
Comment (jeromy-cannon) [inline resources/mirror-node-values-hedera.yaml:1]: instead of creating a clone of the file that could potentially lead to synchronization issues, I would just parse in the file from YAML to JSON, then change the values that need to be changed and write them out to the appropriate cache or staging directory.

File: resources/mirror-node-values.yaml [hunk 1]
```diff
@@ -151,7 +154,7 @@ monitor:
                   transferTypes:
                     - CRYPTO
                 receiptPercent: 1
-                tps: 1
+                tps: 0
```
Comment (jeromy-cannon) [inline resources/mirror-node-values.yaml:157]: what does tps of zero do?

Comment (Ivo-Yankov) [reply resources/mirror-node-values.yaml:157]: It disables the pinger, otherwise it will always be enabled even if the `--pinger` flag isn't passed during `mirror-node deploy`

File: resources/mirror-node-values-hedera.yaml
Comment (Ivo-Yankov) [reply resources/mirror-node-values-hedera.yaml:1]: I think ideally `mirror-node-values-hedera.yaml` won't require any further changes since it is only responsible for mirror-node < v0.130.0. Having them as separate files allows us to add structural changes if needed for future versions.

File: test/e2e/commands/dual-cluster-full.test.ts [hunk 2]
```diff
@@ -31,6 +31,11 @@ const endToEndTestSuite: EndToEndTestSuite = new EndToEndTestSuiteBuilder()
   .withNamespace(testName)
   .withDeployment(`${testName}-deployment`)
   .withClusterCount(2)
+  .withConsensusNodesCount(2)
+  .withLoadBalancerEnabled(true)
```
[the remark this answers is not in the document]
Comment (Ivo-Yankov) [reply test/e2e/commands/dual-cluster-full.test.ts:35]: Initially, I started creating a separate test suite for testing mirror node, but later decided to remove it. For that, I required the option to turn off the load balancer and create a deployment with one consensus node, so I added the functionality to the framework. It will be useful for any other suites that use this framework, so I left it in.

**Review by andrewb1269hg — APPROVED** [review]
Review and approve the two .sh scrips.
